### DISCUSSION TITLE
Allow dynamic cookie path

### DISCUSSION THREAD

**Discussion by pikturdev** [nuxt-community/auth-module#1585] [Ideas & Feature Requests]
### What problem does this feature solve?

In applications employing multiple authentication strategies it should be possible to isolate cookie attributes to a path unique to each strategy.

Given two login pages `/admin/login` and `/customer/login`

When I login as admin in one tab 

```
document.cookie 
auth.strategy=admin
```

and login as customer in another tab 

```
document.cookie 
auth.strategy=customer
```

then return to admin tab and refresh

```
document.cookie 
auth.strategy=customer 
```

then token verification fails because the strategy was overridden when we logged in as customer.

### What does the proposed changes look like?

This would cease to be a problem if it were possible to configure the cookie path per strategy or simply allow the developer to provide a function to `options.cookie.options.path` as in the following example.

```js
// nuxt.config.js
auth: {
  cookie: {
    options: {
      path: () => /path/.test(window.location.pathname) ? '/path-a' : '/path-b'
    }
  }
}

// auth/storage.js
setCookie (key, value, options = {}) {
  // ...

  if (_options.path instanceof Function) {
    _options.path = _options.path()
  }

  // ...
}
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c309">#c309</a>)</em></sub></div>

**Comment by kmorpex** [comment]
Found a solution?
